src/components/TodoItem.tsx

```diff
@@ -0,0 +1,66 @@
+import cn from 'classnames';
+import { useState } from 'react';
+import { useTodoContext } from '../context/TodoContext';
+import { deleteTodo } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { Error } from '../types/Error';
+
+interface Props {
+  key: number;
+  todo: Todo;
+}
+
+export const TodoItem: React.FC<Props> = ({ key, todo }) => {
+  const { setTodos, setError } = useTodoContext();
+  const [isTodoDeleting, setIsTodoDeleting] = useState(false);
+
+  const handleTodoDelete = async () => {
+    setIsTodoDeleting(true);
+
+    try {
+      await deleteTodo(todo.id);
+
+      setTodos((prevTodos) => prevTodos.filter((t) => t.id !== todo.id));
+      setIsTodoDeleting(false);
+    } catch (error) {
+      setError(Error.DELETE);
```

Comment:
It is better to use the error object from the catch block to set a more specific error message. Change 'setError(Error.DELETE)' to 'setError(error.message || Error.DELETE)'.